src/time_freezing_reformulation.m

```diff
@@ -244,18 +245,27 @@
         f_aux_pos = []; % matrix wit all aux tan dyn
         f_aux_neg = [];
         % time freezing dynamics
-        f_aux_normal = [zeros(n_q,n_contacts);inv_M_aux*J_normal*a_n;zeros(1,n_contacts)];
+        f_q_dynamics = zeros(n_q,1);
+        if settings.stabilizing_q_dynamics
+            f_aux_normal = [-settings.kappa_stabilizing_q_dynamics*J_normal.*(f_c);inv_M_aux*J_normal*a_n;zeros(1,n_contacts)];
+        else
+            f_aux_normal = [f_q_dynamics;J_normal*a_n;zeros(1,n_contacts)];
```

Comment:
I would probably just use `zeros(n_q,1)` here and only define `f_q_dynamics` in the loop or set `f_q_dynamics` in an if statement and only assign `f_aux_normal` once.